src/test/ui/async-await/in-trait/async-associated-types2-desugared.rs

```diff
@@ -0,0 +1,31 @@
+// edition: 2021
+
+#![feature(async_fn_in_trait)]
+#![allow(incomplete_features)]
+
+use std::future::Future;
+
+trait MyTrait {
+    type Fut<'a>: Future<Output = i32>
+    where
+        Self: 'a;
+
+    async fn foo(&self) -> Self::Fut<'a>;
```

Comment:
Can you fix this? This should probably be `&'a self` (and adding `'a` generic lifetime) or `Self::Fut<'_>`.

Reply:
```suggestion
    fn foo(&self) -> Self::Fut<'a>;
```

I'm not 100% sure what this test is trying to test, but the name "desugared" suggests to me that it's trying to write out explicitly the equivalent of an `async fn`

Reply:
> Can you fix this? This should probably be `&'a self` (and adding `'a` generic lifetime) or `Self::Fut<'_>`.

oh, yeah that makes sense. I'll update it

Reply:
@spastorino what did you have in mind for this one (example #10 in your HackMD)?I know what desugared means in a general sense, not sure what that would look like for this though.

Reply:
Actually, I understand now, after re-reading the static AFIT RFC